_posts/2017-06-06-Spring-IO-2017-The-Spring-ecosystem.md

```diff
@@ -0,0 +1,318 @@
+It makes sure all the connections are created and correctly closed. 
+This library helps out with the looping through the results and filtering those.
+It's also possible to manage your transactions with a client-side transaction manager. 
+If you're working with this Lightweight Data Access Protocol, this might definitely be worth your while.
+
+## Spring Integration
+
+<span class="image left small"><img class="p-image" alt="Spring Integration" src="/img/spring-ecosystem/spring-integration.png"></span>
+
+When an architecture is focused around events or messages, you can get the help of Spring Integration. 
+This resolves around the implementation of Enterprise Integration patterns. 
+When you want to send something from point A to point B, there could be a lot of different network protocols or restrictions in between. 
+Spring Integration minimizes the boilerplate code needed by implementing those patterns. 
+It just makes it easy to send events and messages throughout different endpoints.
+
+## Spring Batch
+
+<span class="image right small"><img class="p-image" alt="Spring Batch" src="/img/spring-ecosystem/spring-batch.png"></span>
+
+With Spring Batch it is possible to write an offline batch application using Java and Spring.
+It makes it very convenient when you're used to the Spring Framework to execute a bunch of jobs.
+It features a possibility to read and write your resource and a way of dividing data for processing and much more.
+But also supports a transaction manager, job processing statistics, job status changes and much more.
+
+## Spring Web Flow
+
+<span class="image left small"><img class="p-image" alt="Spring Web Flow" src="/img/spring-ecosystem/spring-web-flow.png"></span>
+
+The Spring Web Flow is created to help users to navigate through the different views of a stateful web application. 
+A common example could be when shopping online. 
+The process has a clear starting and finishing view, but in between it can change state or views dynamically. 
+Through guided navigations, the user makes changes and it should register those changes as well as a possibility to finalise those change through a confirmation. 
+All this is possible with Spring Web Flow. 
+Although this project is listed with the main projects, there hasn't been any progress over the last years.
+
+## Spring Web Services
+
+<span class="image right small"><img class="p-image" alt="Spring Web Services" src="/img/spring-ecosystem/spring-web-services.png"></span>
+
+There are several ways to develop a web service, and one of which is used in combination with SOAP. 
+Spring Web Services helps with creating contract-first SOAP web services which are flexible by manipulating the XML contents. 
+But due to the popularity of the architectural style of REST, the interest in SOAP has diminished and that is noticeable in the maintenance of this Spring project which hasn't had any significant version updates. 
```

Comment:
can be interesting to add when the lastest significant update was (I presume it has been a while)